kube/pod.go

```diff
@@ -35,19 +35,41 @@ func NewPodTemplate(role *model.Role, settings ExportSettings, grapher util.Mode
 	}
 
 	var resources helm.Node
+	var requests *helm.Mapping
+	var limits *helm.Mapping
+
+	if settings.UseMemoryLimits || settings.UseCPULimits {
+		requests = helm.NewMapping()
+		limits = helm.NewMapping()
+		resources = helm.NewMapping("requests", requests, "limits", limits)
+	}
+
 	if settings.UseMemoryLimits {
 		if settings.CreateHelmChart {
-			reqMap := helm.NewMapping("memory", fmt.Sprintf("{{ int .Values.sizing.%s.memory }}Mi", roleVarName))
-			limMap := helm.NewMapping("memory", fmt.Sprintf("{{ (mul (int .Values.sizing.memory.limit_factor) (int .Values.sizing.%s.memory)) }}Mi", roleVarName))
+			requests.Add("memory",
+				helm.NewNode(fmt.Sprintf("{{ int .Values.sizing.%s.memory }}Mi", roleVarName),
+					helm.Block("if .Values.sizing.memory.requests")))
 
-			reqMap.Set(helm.Block("if .Values.sizing.memory.requests"))
-			limMap.Set(helm.Block("if .Values.sizing.memory.limits"))
+			limits.Add("memory",
+				helm.NewNode(fmt.Sprintf("{{ (mul (int .Values.sizing.memory.limit_factor) (int .Values.sizing.%s.memory)) }}Mi", roleVarName),
+					helm.Block("if .Values.sizing.memory.limits")))
+		} else {
+			requests.Add("memory", fmt.Sprintf("%dMi", role.Run.Memory))
+			limits.Add("memory", fmt.Sprintf("%dMi", settings.MemLimitFactor*role.Run.Memory))
+		}
+	}
+	if settings.UseCPULimits {
+		if settings.CreateHelmChart {
+			requests.Add("cpu",
+				helm.NewNode(fmt.Sprintf("{{ int .Values.sizing.%s.vcpu_count }}m", roleVarName),
+					helm.Block("if .Values.sizing.cpu.requests")))
 
-			resources = helm.NewMapping("requests", reqMap, "limits", limMap)
+			limits.Add("cpu",
```

Comment:
Can you add a comment explaining what this yields given some input?

Reply:
You mean how the generated template looks like ? Can do. Has to wait a bit, currently set to testing Jan's port templating, with a different fissile.